### ISSUE TITLE
Process endpoints in parallel

### ISSUE THREAD

**Issue by ppf2** [logstash-plugins/logstash-input-http_poller#47]
Currently, the poller goes through each url/endpoint serially.  If there are many (thousands) of endpoints to go through, it will be nice to have some kind of thread pool implemented so that it can process concurrent endpoints in parallel.

**Comment by suyograo** [comment]
+1, good idea
